lib/ansiblelint/rules/RoleNames.py

```diff
@@ -21,15 +21,19 @@
 
 import re
 from pathlib import Path
-from typing import List
+from typing import TYPE_CHECKING, List
 
+from ansiblelint.errors import MatchError
 from ansiblelint.rules import AnsibleLintRule
 from ansiblelint.utils import parse_yaml_from_file
 
+if TYPE_CHECKING:
+    from ansiblelint.file_utils import TargetFile
+
 ROLE_NAME_REGEX = '^[a-z][a-z0-9_]+$'
 
 
-def _remove_prefix(text, prefix):
+def _remove_prefix(text: str, prefix: str) -> str:
     return re.sub(r'^{0}'.format(re.escape(prefix)), '', text)
```

Comment:
This looks extremely inefficient. Why not simply do:
```suggestion
    return text[len(prefix):] if text.startswith(prefix) else text
```

Reply:
Outside the scope of this change, but we can it separated. In fact the only reason I touched it was because adding type hints to the other methods required type of it too.